src/dds/rs-dds-auto-calibration.cpp

```diff
@@ -0,0 +1,96 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+
+#include "rs-dds-auto-calibration.h"
+#include <rsutils/string/from.h>
+#include <rsutils/json.h>
+#include <src/ds/d500/d500-auto-calibration.h>
+
+
+namespace librealsense
+{
+    dds_auto_calibrated::dds_auto_calibrated() :
+        _d500_ac(std::make_shared<d500_auto_calibrated>()) {}
+
+
+    std::vector<uint8_t> dds_auto_calibrated::run_on_chip_calibration(int timeout_ms, std::string json, 
+        float* const health, rs2_update_progress_callback_sptr progress_callback)
+    {
+        return _d500_ac->run_on_chip_calibration(timeout_ms, json, health, progress_callback);
+    }
+
+
+    std::vector<uint8_t> dds_auto_calibrated::run_tare_calibration(int timeout_ms, float ground_truth_mm, std::string json, float* const health, rs2_update_progress_callback_sptr progress_callback)
+    {
+        throw std::runtime_error(rsutils::string::from() << "Tare Calibration not applicable for this device");
+    }
+
+    std::vector<uint8_t> dds_auto_calibrated::process_calibration_frame(int timeout_ms, const rs2_frame* f, float* const health, rs2_update_progress_callback_sptr progress_callback)
+    {
+        throw std::runtime_error(rsutils::string::from() << "Process Calibration Frame not applicable for this device");
+    }
+
+    std::vector<uint8_t> dds_auto_calibrated::get_calibration_table() const
+    {
+        throw std::runtime_error(rsutils::string::from() << "Get Calibration Table not applicable for this device");
+    }
+
+    void dds_auto_calibrated::write_calibration() const
+    {
+        throw std::runtime_error(rsutils::string::from() << "Write Calibration not applicable for this device");
+    }
+
+    void dds_auto_calibrated::set_calibration_table(const std::vector<uint8_t>& calibration)
+    {
+        throw std::runtime_error(rsutils::string::from() << "Set Calibration Table not applicable for this device");
+    }
+
+    void dds_auto_calibrated::reset_to_factory_calibration() const
+    {
+        throw std::runtime_error(rsutils::string::from() << "Tare Calibration not applicable for this device");
```

Comment:
Tare?

Reply:
typo - thanks
BTW, we may need to implement also this method later on - don't you think?

Reply:
Could be..